### PR TITLE
fix(core): extract named enums inside nullable nested objects

### LINKED ISSUE #3340: Nested object with enum value no longer exports enum as const
### Description

Hi, it seems that somewhere since orval 8.0.0 there's been a change in the way enum values in a nested object are generated by Orval.

In 7.21.0, the values would be extracted to a `const ... = {...} as const` construction, but starting in 8.0.0, the values are inlined as string values.

### Output client

react-query

### Configuration (orval.config)

```ts
const {defineConfig} = require("orval");

module.exports = defineConfig({
    "test": {
        "input": {
            target: "./src/openapi-test.yaml"
        },
        output: {
            clean: true,
            workspace: "./src/data/test",
            target: "./generated/test.ts",
            schemas: "./generated/schemas",
            client: "react-query",
            mode: "tags-split"
        }
    }
});

```

### Environment

  System:
    OS: macOS 26.4.1
    CPU: (12) arm64 Apple M2 Max
    Memory: 67.31 MB / 64.00 GB
    Shell: 5.9 - /bin/zsh
  npmPackages:
    orval: 8.2.0 => 8.2.0 

### Expected behavior

This is the output in v7.21.0:

```typescript
export type TestMonthSelection = {
  /**
   * @minItems 1
   * @maxItems 12
   */
  months: TestMonthSelectionMonthsItem[];
  months2?: TestMonthSelectionMonths2;
} | null;
```

```typescript
// TestMonthSelectionMonthsItem.ts
export type TestMonthSelectionMonthsItem = typeof TestMonthSelectionMonthsItem[keyof typeof TestMonthSelectionMonthsItem];


// eslint-disable-next-line @typescript-eslint/no-redeclare
export const TestMonthSelectionMonthsItem = {
  JANUARY: 'JANUARY',
  FEBRUARY: 'FEBRUARY',
  MARCH: 'MARCH',
  APRIL: 'APRIL',
  MAY: 'MAY',
  JUNE: 'JUNE',
  JULY: 'JULY',
  AUGUST: 'AUGUST',
  SEPTEMBER: 'SEPTEMBER',
  OCTOBER: 'OCTOBER',
  NOVEMBER: 'NOVEMBER',
  DECEMBER: 'DECEMBER',
} as const;
```
I would at least like a configuration option to allow this type of export to be possible.

### Actual behavior

This is the output post v8.0.0:
```typescript
export type TestMonthSelection = {
  /**
   * @minItems 1
   * @maxItems 12
   */
  months: ('JANUARY' | 'FEBRUARY' | 'MARCH' | 'APRIL' | 'MAY' | 'JUNE' | 'JULY' | 'AUGUST' | 'SEPTEMBER' | 'OCTOBER' | 'NOVEMBER' | 'DECEMBER')[];
  months2?: 'JANUARY' | 'FEBRUARY' | 'MARCH' | 'APRIL' | 'MAY' | 'JUNE' | 'JULY' | 'AUGUST' | 'SEPTEMBER' | 'OCTOBER' | 'NOVEMBER' | 'DECEMBER';
} | null;
```

### OpenAPI document (minimal, if applicable)

```yaml
openapi: 3.0.0
info:
  title: OpenAPI Test
  version: 1.0.0
components:
  schemas:
    test:
      type: object
      required:
        - months
      properties:
        name:
          enum:
            - option1
            - option2
            - option3
            - option4
          nullable: true
        months:
          maxItems: 12
          minItems: 1
          uniqueItems: true
          type: array
          items:
            type: string
            enum:
              - JANUARY
              - FEBRUARY
              - MARCH
              - APRIL
              - MAY
              - JUNE
              - JULY
              - AUGUST
              - SEPTEMBER
              - OCTOBER
              - NOVEMBER
              - DECEMBER
        MonthSelection:
          nullable: true
          required:
            - months
          type: object
          properties:
            months:
              maxItems: 12
              minItems: 1
              uniqueItems: true
              type: array
              items:
                type: string
                enum:
                  - JANUARY
                  - FEBRUARY
                  - MARCH
                  - APRIL
                  - MAY
                  - JUNE
                  - JULY
                  - AUGUST
                  - SEPTEMBER
                  - OCTOBER
                  - NOVEMBER
                  - DECEMBER
            months2:
              type: string
              enum:
                - JANUARY
                - FEBRUARY
                - MARCH
                - APRIL
                - MAY
                - JUNE
                - JULY
                - AUGUST
                - SEPTEMBER
                - OCTOBER
                - NOVEMBER
                - DECEMBER

paths:
  /api/test:
    get:
      responses:
        '200':
          description: Successful response
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/test'
```

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by wadakatu**
## Problem

Closes #3340.

Since v8.0.0, `enum` values declared inside a **nullable** nested object are inlined as string-literal unions instead of being extracted into a named `const … = { … } as const`. Top-level properties and **non-nullable** nested objects still extract correctly, so the behavior is inconsistent.

```ts
// before (v8.x)
export type TestMonthSelection = {
  months: ('JANUARY' | 'FEBRUARY' | 'MARCH')[];   // ❌ inlined
  months2?: 'JANUARY' | 'FEBRUARY' | 'MARCH';      // ❌ inlined
} | null;

// after (matches v7.21.0)
export type TestMonthSelection = {
  months: TestMonthSelectionMonthsItem[];          // ✅ named `as const`
  months2?: TestMonthSelectionMonths2;             // ✅ named `as const`
} | null;
```

## Root cause

A `nullable: true` object is upgraded to `type: ['object', 'null']`, which makes `getObject` route it through `combineSchemas` instead of the normal object property-iteration path. With the v8 default (`aliasCombinedTypes: false`), `combineSchemas` resolves the object variant with an **undefined `propName`**, so the inner `getObject` receives no `name`. Without a name, nested enum properties can't build the identifier they'd be extracted under (`if (propName && resolvedValue.isEnum && …)` in `resolvers/object.ts`), so they fall back to inline unions.

A non-nullable object keeps `type: 'object'` (a string), stays on the property-iteration path with `name` intact, and extracts correctly — hence the asymmetry.

## Fix

In `getObject`'s type-array branch, detect a nullable object (`type` array whose only non-null entry is `object`, **with** properties) and let it fall through to the existing property-iteration path instead of `combineSchemas`. This preserves `name` so nested enums extract as before; `nullable` already carries the trailing ` | null`.

The guard is deliberately narrow:
- **Real unions** (e.g. `['string', 'number']`) → unchanged (`combineSchemas`).
- **Empty nullable objects** (no properties) → unchanged (`combineSchemas`), so they keep rendering as `{ [key: string]: unknown } | null` rather than `unknown`.
- Only nullable objects **with** properties — the buggy case — change.

No new config option: this just restores the consistent extract-everywhere behavior.

## Tests & verification

- Added a `generateSchemasDefinition` regression test (`schema-definition.test.ts`) covering the `type: ['object','null']` shape with an enum array + scalar enum.
- `@orval/core` unit tests: 1971 passing.
- Full snapshot suite (`test:snapshots`, samples + orval-tests): passing, **no changes to any committed sample output**.
- `typecheck` + `lint`: passing.
- Generated output for the issue's spec now matches v7.21.0. Verified a broad edge-case spec (real unions, nullable scalars/arrays, `$ref` children, deep nesting, `additionalProperties`, top-level nullable component, nullable object inside arrays, empty nullable object) compiles cleanly under `tsc --strict`.

## Out of scope

The OAS 3.1 explicit `anyOf`/`oneOf: [{ object }, { type: null }]` spelling routes through a different (combiner) branch and still inlines its child enums. Happy to follow up on that separately.

## Summary by CodeRabbit

* **Bug Fixes**
  * Fixed enum extraction for properties inside nullable nested objects so enums are generated as named constants and nullable types include `null`.

* **Tests**
  * Added a regression test to verify correct enum extraction and naming for nested nullable object schemas.

### CODE REVIEW

File: packages/core/src/getters/object.ts [hunk 1]
```diff
@@ -223,20 +223,44 @@ export function getObject({
...
+      | Record<string, unknown>
+      | undefined;
+    // Only divert when there are properties to walk — an empty nullable object
+    // has no nested enums to extract, and routing it through the property path
+    // would render it as `unknown` instead of `{ [key: string]: unknown }`.
+    const isNullableObject =
+      nonNullTypes.length === 1 &&
+      nonNullTypes[0] === 'object' &&
+      typeArrayProperties != null &&
+      Object.keys(typeArrayProperties).length > 0;
```
[the remark this answers is not in the document]
Comment (wadakatu) [reply packages/core/src/getters/object.ts:246]: Good call — fixed in d5bbd5ca9. Cast to `Record<string, OpenApiSchemaObject | OpenApiReferenceObject>` to match the existing `itemProperties` cast a few lines below (the cast is only needed to work around the AnyOtherAttribute `any` infection).

File: packages/core/src/generators/schema-definition.test.ts [hunk 2]
```diff
@@ -305,6 +305,70 @@ describe('generateSchemasDefinition', () => {
...
+    const monthSelection = result.find((s) => s.name === 'TestMonthSelection');
+    expect(monthSelection).toBeDefined();
+    expect(monthSelection?.model).not.toContain(`'JANUARY'`);
+    expect(monthSelection?.model).toMatch(/months2\?: [A-Z]\w+;/);
+    expect(monthSelection?.model).toContain('| null');
+
+    // The nested enums are emitted as standalone `as const` schemas.
+    const constEnums = result.filter((s) => s.model.includes('as const'));
+    expect(constEnums.length).toBeGreaterThanOrEqual(2);
+    expect(constEnums.every((s) => s.model.includes('JANUARY'))).toBe(true);
```
[the remark this answers is not in the document]
Comment (wadakatu) [reply packages/core/src/generators/schema-definition.test.ts]: Addressed in d5bbd5ca9. The test now asserts the specific extracted type names are referenced (`TestMonthSelectionMonthsItem` / `TestMonthSelectionMonths2`) and that each of those named schemas is emitted with `as const`, instead of filtering by an `as const` substring + requiring every match to contain JANUARY. Kept the explicit name assertions intentionally — the generated names are part of the regression contract (they match the v7.21.0 output in the issue).

**Review by melloware — APPROVED** [review]
